asyncwrapper/src/main/kotlin/org/n/riesgos/asyncwrapper/process/wps/WPSProcess.kt

```diff
@@ -1,19 +1,23 @@
 package org.n.riesgos.asyncwrapper.process.wps
 
+import org.n.riesgos.asyncwrapper.config.RetryConfiguration
 import org.n.riesgos.asyncwrapper.config.WPSOutputDefinition
 import org.n.riesgos.asyncwrapper.process.*
 import org.n.riesgos.asyncwrapper.process.Process
 import org.n52.geoprocessing.wps.client.ExecuteRequestBuilder
+import org.n52.geoprocessing.wps.client.WPSClientException
 import org.n52.geoprocessing.wps.client.WPSClientSession
 import org.n52.geoprocessing.wps.client.encoder.WPS20ExecuteEncoder
 import org.n52.geoprocessing.wps.client.model.*
 import org.n52.geoprocessing.wps.client.model.execution.Data
-import java.sql.Ref
 import java.util.*
 import java.util.logging.Logger
 
 
-class WPSProcess(private val wpsClient : WPSClientSession, private val url: String, private val processID: String, private val wpsVersion: String, private val expectedOutputs : List<WPSOutputDefinition>) : Process {
+class WPSProcess(private val wpsClient : WPSClientSession, private val url: String, private val processID: String, private val wpsVersion: String, private val expectedOutputs : List<WPSOutputDefinition>, private val retryConfig : RetryConfiguration) : Process {
+
+    private var retryAttempts = 0;
```

Comment:
As far as possible, I would not make the retry count a local variable. Instead I would prefer to have it local in the methods themselves, so we don't have to care about its current state in the WPSProcess instance itself.